src/models/SiteBanner.php

```diff
@@ -190,13 +190,16 @@ public function isActive()
         if (!$config->embargo_enabled) {
             return true;
         }
+        if ($this->StartDate && $this->EndDate) {
+            $startDate = new DateTime($this->StartDate);
+            $endDate   = new DateTime($this->EndDate);
+            $now       = new DateTime(DBDatetime::now()->Format(DBDatetime::ISO_DATETIME));
 
-        $startDate = new DateTime($this->StartDate);
-        $endDate   = new DateTime($this->EndDate);
-        $now       = new DateTime(DBDatetime::now()->Format(DBDatetime::ISO_DATETIME));
+            // Check if the current time falls between the start and end dates.
+            return ((!$this->StartDate || $startDate <= $now) && (!$this->EndDate || $endDate >= $now));
```

Comment:
Checking `!$this->StartDate` or `!$this->EndDate` here is useless since it's within a block that mandates both to be truthy, so those conditions can be removed and this should be just
```
return ($startDate <= $now) && ($endDate >= $now));
```